modules/search/instant-search/components/search-app.jsx

```diff
@@ -61,56 +60,91 @@ class SearchApp extends Component {
 		} );
 	}
 
+	componentDidMount() {
+		if ( this.props.grabFocus ) {
+			this.input.current.focus();
+		}
+	}
+
+	hasActiveQuery() {
+		return this.state.query !== '' || hasFilter();
+	}
+
+	activateResults() {
+		if ( ! this.state.resultsActive ) {
+			this.setState( { resultsActive: true } );
+			hideChildren( this.props.themeOptions.results_selector );
+		}
+	}
+
+	maybeDeactivateResults() {
+		if ( this.isSearchPage || this.hasActiveQuery() ) {
+			return;
+		}
+		if ( this.state.resultsActive ) {
+			this.setState( { resultsActive: false } );
+			showChildren( this.props.themeOptions.results_selector );
+		}
+	}
+
+	onSearchFocus = () => {
+		this.activateResults();
+	};
+
+	onSearchBlur = () => {
+		this.maybeDeactivateResults();
+	};
+
 	onChangeQuery = event => {
+		this.activateResults();
 		const query = event.target.value;
 		this.setState( { query } );
 		setSearchQuery( query );
-		this.getResults( query, getFilterQuery(), getSortQuery() );
+		this.getDebouncedResults( query, getFilterQuery(), getSortQuery() );
 	};
 
 	onChangeFilter = ( filterName, filterValue ) => {
 		setFilterQuery( filterName, filterValue );
+		if ( this.hasActiveQuery() ) {
+			this.activateResults();
+		}
+		this.maybeDeactivateResults();
```

Comment:
Shouldn't this line be in an `else` clause?

Reply:
It doesn't actually matter because of then if clause in `maybeDeactivateResults()` but I'll add one to clarify the code.